src/json.rs

```diff
@@ -0,0 +1,426 @@
+// Vendored from `smoljson` bef592c5da1c3fe38b2462a8d231b0e0c8a86f80, with explicit permission
+// from the author (`thomcc`). Minimized for cc/simplicity. Modifications and additions made to fit cc's needs.
+
+use std::borrow::Cow;
+
+#[derive(PartialEq, Debug, Clone)]
+pub(crate) enum Token<'s> {
+    Null,
+    Bool(bool),
+    NumU(u64),
```

Comment:
You can just use Num(f64), I think. The i64/u64 is because I had a special use for that code.